apps/myjobs/app/models/workflow.rb

```diff
@@ -124,14 +124,46 @@ def stage
     Pathname.new(self.staged_dir)
   end
 
-  # Get an array of all the files of a directory
+  # Get an array of WorkflowFile Objects of all the files of a directory
   #
   # Find.find returns an enumerator - the first path is always the initial directory
   # so we return the array with the first item omitted
+  # Then, map files to WorkflowFile objects
+  #
+  # @return [WorkflowFile] An array of WorkflowFile Objects of all the files of a directory
   def folder_contents
```

Comment:
could cache this like:

```ruby
def folder_contents
  return @folder_contents if defined? @folder_contents

  ...
end
```

Reply:
This actually only works if you set the @folder_contents otherwise. For example:

```ruby
def folder_contents
  return @folder_contents if defined? @folder_contents

  if File.directory?(self.staged_dir)
    @folder_contents = Find.find(self.staged_dir).drop(1).select {|f| File.file?(f) }.map {|f| WorkflowFile.new(f, self.staged_dir)}
  else
    @folder_contents = []
  end
end
```